### PR TITLE
Add docs for Alluxio file system

### PR DESCRIPTION
**Pull request by jja725**
## Description

Add docs for Alluxio file system.

## Additional context and related issues

Docs for https://github.com/trinodb/trino/pull/21603

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/object-storage.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ compatible replacements:
 * [](/object-storage/file-system-azure)
 * [](/object-storage/file-system-gcs)
 * [](/object-storage/file-system-s3)
+* [](/object-storage/file-system-alluxio)
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage.md:63]: Not sure if we want it here .. wdyt @electrum ?

File: docs/src/main/sphinx/object-storage/file-system-alluxio.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - Description
+* - `fs.alluxio.enabled`
+  - Activate the native implementation for Alluxio Filesystem support. Defaults to `false`.
+:::
+
+## Alluxio client configuration
+
+The Alluxio client in `alluxio-site.properties` configures the connection to the
+Alluxio cluster. The default location of the `alluxio-site.properties` is
+`/opt/alluxio/conf`. Follow the [Alluxio client configuration
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-alluxio.md:34]: Is this location configurable .. if yes .. we should add to the table. Also .. I assume this file is needed on the Trino coordinator and all workers .. right?

**Comment by mosabua** [comment]
I did a bunch of updates for grammar and setup so the docs actually build and work. Left a few questions but in general this looks good. I am wondering if we should have a minimal example of the properties file though .. and then link to the Alluxio docs for me. 

Also please review all content I updated and update as needed and push further changes @jja725

File: docs/src/main/sphinx/object-storage/file-system-alluxio.md (hunk 2 above)
Comment (jja725) [reply docs/src/main/sphinx/object-storage/file-system-alluxio.md:34]: The location is only configurable through Trino side at the moment. We can make it configurable in a later PR. Yes it's need to be on every process.

**Comment by jja725** [comment]
> I did a bunch of updates for grammar and setup so the docs actually build and work. Left a few questions but in general this looks good. I am wondering if we should have a minimal example of the properties file though .. and then link to the Alluxio docs for me.
> 
> Also please review all content I updated and update as needed and push further changes @jja725

Added example property file. The updates LGTM, thanks for collaborating on this!

File: docs/src/main/sphinx/object-storage/file-system-alluxio.md (hunk 2 above)
Comment (jja725) [reply docs/src/main/sphinx/object-storage/file-system-alluxio.md]: sorry for the typo, the location is not configurable at the moment. We can make it configurable at Trino side in a later PR

Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-alluxio.md]: Sounds good

File: docs/src/main/sphinx/object-storage/file-system-alluxio.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* - Property
+  - Description
+* - `fs.alluxio.enabled`
+  - Activate the Alluxio file system support. Defaults to `false`.
+:::
+
+## Alluxio client configuration
+
+The Alluxio client in `alluxio-site.properties` configures the connection to the
+Alluxio cluster. The default location of the `alluxio-site.properties` is
```
[the remark this answers is not in the document]
Comment (jja725) [reply docs/src/main/sphinx/object-storage/file-system-alluxio.md]: resolved

**Review by mosabua — APPROVED** [review]
Looks good now. Thank you @jja725

**Pull request merged**